tools/snp.sh

```diff
@@ -164,6 +163,72 @@ verify_snp_host() {
   fi
 }
 
+verify_platform_snp_bit_status() {
+  if [ "$1" == "host" ]; then
+    # Get the host cpuid eax
+    local host_cpuid_eax
+    host_cpuid_eax=$(get_cpuid 0x8000001f eax)
+
+    # Map all the computed host sev/snp bit values in a single associative array
+    declare -A actual_sev_snp_bit_status=(
+      [SME]=$(( (${host_cpuid_eax} >> 0) & 1))
+      [SEV]=$(( (${host_cpuid_eax} >> 1) & 1))
+      [SEV-ES]=$(( (${host_cpuid_eax} >> 3) & 1))
+      [SNP]=$(( (${host_cpuid_eax} >> 4) & 1))
+    )
+
+  fi
+
+  if [ "$1" == "guest" ]; then
+    if [ ! -f "${GUEST_SSH_KEY_PATH}" ]; then
+      >&2 echo -e "Guest SSH key not present [${GUEST_SSH_KEY_PATH}], so cannot verify guest SNP enabled"
+      return 1
+    fi
+
+    # Install guest rdmsr package dependencies to insert & insert guest msr module
+    ssh_guest_command "sudo DEBIAN_FRONTEND=noninteractive sudo apt install -y msr-tools > /dev/null 2>&1" > /dev/null 2>&1
+    ssh_guest_command "sudo modprobe msr" > /dev/null 2>&1
```

Comment:
Will this only work on Debian?

Reply:
Oh.. good catch, yes this works only on debian